perception/image_projection_based_fusion/src/fusion_node.cpp

```diff
@@ -53,87 +56,67 @@ FusionNode<Msg, ObjType>::FusionNode(
     rois_number_ = 8;
   }
 
-  // subscribers
-  sub_.subscribe(this, "input", rclcpp::QoS(1).get_rmw_qos_profile());
+  // Set parameters
+  match_threshold_ms_ = static_cast<int>(declare_parameter("match_threshold_ms", 50));
+  timeout_ms_ = static_cast<double>(declare_parameter("timeout_ms", 0.1));
+  input_offset_ms_ = declare_parameter("input_offset_ms", std::vector<double>{});
+  if (!input_offset_ms_.empty() && rois_number_ != input_offset_ms_.size()) {
+    RCLCPP_ERROR(get_logger(), "The number of topics does not match the number of offsets.");
+    return;
+  }
```

Comment:
```suggestion
  if (input_offset_ms_.empty() || rois_number_ != input_offset_ms_.size()) {
    throw std::runtime_error("The number of offsets does not match the number of topics.");
  }
```

Reply:
May I ask what is the difference?

Reply:
The constructor just exits in your code but we want the program to exit with error if rois_number and the size of offsets don't matches.